MetaMask/metamask-extension
# NetworkController: Use messenger for events

## Explanation



Currently, the network controller notifies consumers about events by emitting them directly from the controller. In order to migrate the controller to the core repo, where controllers use the BaseControllerV2 interface, events should be emitted via a messenger object.

This commit updates the network controller to use a messenger, and then updates all of the controllers that listen for network events to use the messenger as well.

Closes #16945.

For more context behind this change, see: https://github.com/MetaMask/core/issues/971

## Manual Testing Steps



There were a few changes to controllers other than the network controller, but most of the behavior affected centers around what happens when you switch to a different network. I would recommend these tests:

* When Infura is blocked, a blocked message should appear
* Swaps functionality should work before and after switching to a different network
* Metrics events should include the chain ID of the current network, even when switching to a different network
* When sending a transaction, gas fees should be correct for the current network, even when switching to a different network
* Should be able to send a transaction from one account to another, i.e. an incoming transaction should appear in the new account's list automatically, and this should work even after switching to a different network and issuing a send there
* The currency rate (shown when sending a transaction?) should reflect the current network, even after switching to a different one
* The balance of all accounts within the current network should be correct, even after switching to a different network
* When switching to a different network, any unapproved (pending?) transactions should disappear

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
